File: dsl-style-java/src/test/java/example/dsl_style_java/BatchInsertTest.java
package example.dsl_style_java;

import example.common.domain.Age;
import example.common.domain.Salary;
import example.common.entity.Employee;
import example.common.test.TestEnvironment;
import example.dsl_style_java.repository.EmployeeRepository;
import java.util.Arrays;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.seasar.doma.jdbc.Config;

@ExtendWith(TestEnvironment.class)
public class BatchInsertTest {

  private final EmployeeRepository repository;

  BatchInsertTest(Config config) {
    repository = new EmployeeRepository(config);
  }

  @Test
  public void testBatchInsert() {
    Employee employee1 = new Employee();
    employee1.setName("test-1");
    employee1.setAge(new Age(30));
    employee1.setSalary(new Salary(300));

    Employee employee2 = new Employee();
    employee2.setName("test-2");
    employee2.setAge(new Age(40));
    employee2.setSalary(new Salary(500));

    repository.batchInsert(Arrays.asList(employee1, employee2));
  }
}

File: dsl-style-kotlin/src/test/kotlin/example/dsl/style/kotlin/BatchInsertTest.kt
package example.dsl.style.kotlin

import example.dsl.style.kotlin.domain.Age
import example.dsl.style.kotlin.domain.Salary
import example.dsl.style.kotlin.entity.Employee
import example.dsl.style.kotlin.repository.EmployeeRepository
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.seasar.doma.jdbc.Config

@ExtendWith(TestEnvironment::class)
class BatchInsertTest internal constructor(config: Config) {

    private val repository: EmployeeRepository = EmployeeRepository(config)

    @Test
    fun testBatchInsert() {
        val employee1 = Employee()
        employee1.name = "test-1"
        employee1.age = Age(30)
        employee1.salary = Salary(300)
        val employee2 = Employee()
        employee2.name = "test-2"
        employee2.age = Age(40)
        employee2.salary = Salary(500)
        repository.batchInsert(listOf(employee1, employee2))
    }
}
